### ISSUE TITLE
Go linters support

### ISSUE THREAD

**Issue by lexfrei** [bazelbuild/rules_go#1290]
Hello!
I'm working on adding linters to bazel for my company now.
How does this relate to the ideology of your project?
Do you need a PR with this function?

**Comment by ianthehat** [comment]
I would be interested to hear about your approach for linting. 
We have some general plans to allow tools to be written in a way that works on top of Bazel such that they are abstracted away from the actual file layout and can cope with generated sources, which would allow the normal lint tools to just work.
We also have plans for how to run vet like checks during the build, allowing them to prevent build success. We are hoping that we can make this pluggable, and also allow checks to produce messages without failing, which would allow linters to be plugged in as well, but the design is still in early stages.

**Comment by lexfrei** [comment]
In my case i need test rules which takes files and linters list and gives me errors in case of errors.
I can make you some rules like:
```
go_lint_test(
  name = "..."
  srcs = [ ... ]
)
go_vet_test(
  name = "..."
  srcs = [ ... ]
)
```
Give me advice, if any.

**Comment by ianthehat** [comment]
We already have an experimental vet rule, and it's a failed experiment.
Instead we want to add a verify toolchain that's used by the normal go_library rule to run vet like checks in parallel with compilation.
The question is what should the output be, what tool consumes that output in a useful way.

**Comment by lexfrei** [comment]
Well.
You are right. My idea with separated rules is broken.
I'll make some research in the future, but not now.
Feel free to close this issue.

**Comment by ianthehat** [comment]
Okay, I have filed #1292 that I will update if I make any progress on this (not working on it right now, but soon I hope).
In the mean time if you have any suggestions or requirements, please do mention them.
